client/src/Features/MicrosoftLearn/MicrosoftLearnFilterPane.tsx

```diff
@@ -0,0 +1,242 @@
+  const classes = themedClassNames(styles);
+
+  const noFiltersExist = () => {
+    let selectedFilters = filterStore.selectedFilter;
+    return (
+      selectedFilters[FilterType.products].length === 0 &&
+      selectedFilters[FilterType.roles].length === 0 &&
+      selectedFilters[FilterType.types].length === 0 &&
+      selectedFilters[FilterType.levels].length === 0
+    );
+  };
+
+  return useObserver(() => {
+    return (
+      <div>
+        <ActionButton
+          onClick={() => filterStore.resetFilter()}
+          style={{ display: noFiltersExist() ? 'none' : 'block' }}
+          className={classes.clearAll}
+          text="Clear all filters"
+        />
+        <FilterComponent type={FilterType.products} name="Products" />
+        <Separator />
+        <FilterComponent type={FilterType.roles} name="Roles" />
+        <Separator />
+        <FilterComponent type={FilterType.levels} name="Levels" />
+        <Separator />
+        <FilterComponent type={FilterType.types} name="Types" />
+      </div>
+    );
+  });
+};
+
+const FilterPaneSmall = ({ styles }: IStylesOnly<FilterPaneStyles>): JSX.Element | null => {
+  const { filterStore, filteredCatalogContent, isLoadingCatalog } = useStore('microsoftLearnStore');
+
+  const [mainIsOpen, setMainOpen] = useState(false);
+  const [productIsOpen, setProductOpen] = useState(false);
+  const [roleIsOpen, setRoleOpen] = useState(false);
+  const [typeIsOpen, setTypeOpen] = useState(false);
+  const [levelIsOpen, setLevelOpen] = useState(false);
+  const [panelIsOpen, setPanelOpen] = useState(false);
+
+  const classes = themedClassNames(styles);
+
+  type PanelOptions = FilterType.products | FilterType.roles | FilterType.levels | FilterType.types | 'main';
+
+  const setOpen = (options: PanelOptions[]) => {
+    const open = (opt: PanelOptions) => options.includes(opt);
+
+    setMainOpen(open('main'));
+    setProductOpen(open(FilterType.products));
+    setRoleOpen(open(FilterType.roles));
+    setLevelOpen(open(FilterType.levels));
+    setTypeOpen(open(FilterType.types));
+  };
+
+  const backToMainFilters = (props: any, defaultRender: any) => {
+    if (mainIsOpen) {
+      return <>{defaultRender!(props)}</>;
+    }
+    return (
+      <div className={classes.filterPanelHeader}>
+        <ActionButton onClick={event => setOpen(['main'])}>
+          <Icon iconName="Back" />
+          <Text className="backTitle">All Filters</Text>
+        </ActionButton>
+        {defaultRender!(props)}
+      </div>
+    );
+  };
+
+  const getHeader = () => {
+    if (mainIsOpen) return 'Search Filters';
+    else if (productIsOpen) return 'Products';
+    else if (roleIsOpen) return 'Roles';
+    else if (typeIsOpen) return 'Types';
+    else if (levelIsOpen) return 'Levels';
+    return '';
+  };
+
+  const kFormatter = (num: number | bigint | any) => {
+    return Math.abs(num) > 999 ? (Math.abs(num) / 1000).toFixed(1) + 'K' : Math.sign(num) * Math.abs(num);
```

Comment:
Why have we not considered the sign in case of `K`?

Reply:
I was trying something else here! It is obsolete here now, removing the sign check!